Add mobie-utils recipe

constantinpape opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team][1] using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel][2] if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/mobie_utils```) and found it was in an excellent condition.

synapticarbors commented on recipes/mobie_utils/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+
+build:
+  number: 0
+  script: "pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```
Maybe use the canonical `script` entry:
https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L35

synapticarbors commented on recipes/mobie_utils/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+  number: 0
+  script: "pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+  noarch: python
+  entry_points:
```
Can you add tests under the `test`->`commands` section:
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands

synapticarbors commented on recipes/mobie_utils/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+  host:
+    - python >=3.6
+    - pip
+  run:
```
It's a little unclear what the actual runtime dependencies are since they aren't listed in the `setup.py`, but this list doesn't match what's in the upstream repo in the environment file or conda recipe. For example those list `pybdv` but it's not here. I just want to double check that that's ok.

constantinpape replied:
All the other dependencies (e.g. `pybdv`) are pulled in through `cluster_tools` (which is a package I am also maintaining); so I only listed the additional dependencies here.
I am not sure if this is best practice, but I personally prefer this because:
- it's a bit tedious to replicate the dependencies here
- pins can get out of sync if they are updated upstream and not here

Happy to change this if there is a better solution.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/mobie_utils```) and found some lint.

Here's what I've got...


For **recipes/mobie_utils**:

 * The test section contained an unexpected subsection name. test is not a valid subsection name.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/mobie_utils```) and found it was in an excellent condition.

constantinpape commented:
Thanks for the review @synapticarbors. I addressed the comments about the build command and added CLI test commands.
Not sure what to do about the dependencies, see my response to your comment.
